convert2rhel/toolopts.py

```diff
@@ -338,5 +388,52 @@ def warn_on_unsupported_options():
         utils.ask_to_continue()
 
 
+def options_from_config_files(cfg_path=None):
+    """Get the options from config file.
+
+    The order is:
+    1. if filename is set then load from it
+    2. ~/.convert2rhel.ini
+    3. /etc/convert2rhel.ini
+
+    Return dict of supported options with their values.
+    """
+    headers = ["subscription_manager"]  # supported sections in config file
+    config_file = configparser.ConfigParser()
+    paths = [os.path.expanduser(p) for p in CONFIG_PATHS]
+    # Create dict with all supported options, all of them set to None
+    # needed for avoiding problems with files priority
+    # The name of supported option MUST correspond with the name in ToolOpts()
+    # Otherwise it won't be used
+    supported_opts = {"password": None, "activation_key": None}
+
+    if cfg_path:
+        cfg_path = os.path.expanduser(cfg_path)
+        if not os.path.exists(cfg_path):
+            raise OSError(2, "No such file or directory: '" + cfg_path + "'")
+        paths.insert(0, cfg_path)  # highest priority
+
+    for path in paths:
+        if os.path.exists(path):
+            if not oct(os.stat(path).st_mode)[-4:].endswith("00"):
```

Comment:
What do you think about changing this and instead of checking against permission that ends with `00`, checking that the `st_mode` is returning a `0600` as expected?

```suggestion
            if int(oct(os.stat(path).st_mode)[-4:]) != 0600:
```
An example to illustrate that: 

```bash
touch b
chmod 0700
```

This file should pass on this statement right now since the permission ends on `00`

```python
>>> oct(os.stat('b').st_mode)[-4:]
'0700'
>>> not oct(os.stat('b').st_mode)[-4:].endswith("00")
False
```

Actually, any other permission that ends on `00` should pass on this, and since the message in the `logging.critical()` specifically tells the user that the permissions should be `0600`, it feels misleading looking at the code.

Reply:
@r0x0d that was already one of the ideas. But according to [this](https://github.com/oamg/convert2rhel/pull/451#discussion_r833005319) discussion it's done like this. In short it's because the compatibility between Python 2 & 3.